core/shared/src/main/scala/org/http4s/package.scala

```diff
@@ -81,4 +81,5 @@ package object http4s {
   /** A stream of server-sent events */
   type EventStream[F[_]] = Stream[F, ServerSentEvent]
 
+  type PureF[A] = Nothing
```

Comment:
Sorry to be re-litigating this 😆 but what's the purpose of `PureF` vs just using `Nothing`? Unless it's a technical thing, IIUC since it's a type alias compile errors will just show `Nothing` anyway, right?

Reply:
It's because of https://github.com/lampepfl/dotty/issues/14640 - Scala 3 gets confused with about kinds with `Nothing`. I don't like how this is currently done in this PR though - I'm tempted to revert to just using `fs2.Pure`. I'm also having second thoughts about the type aliases for `Response.Pure` etc - I'm thinking we should maybe just use `Response[fs2.Pure]. Any thoughts on that?

Reply:
Well, since the `F[_]` is the parameter for `fs2`, it makes sense to use the type alias of `fs2`.

Reply:
Gotcha, thanks for the pointer. Yeah, maybe let's dial back the aliases for now. This PR does a great thing at a high level (I wanna land it!) and at the moment these type issues just seem a bit in the mud.

Reply:
👍 I'm also thinking about removing the existing `EntityEncoder.Pure` alias - what do you think?

Reply:
@bplommer Good enough is good enough, and the changes in this PR are relevant. Bike-shedding on type aliases may wait.